### PR TITLE
🚸 Improve the user experience for setting up Python & reticulate

### PR DESCRIPTION
**Pull request by lazappi**
**Related to:** #128 

## Description

- Add a `install_lamindb()` function
- Add a `.onLoad()` function that sets the default Python environment
- Move **{reticulate}** to `Imports`
- Add a `lamin_connect()` function that calls `lamin connect`
- Add a `lamin_login()` function that calls `lamin login`
- Add a set up vignette and updates other docs to use new functions
- Update CI to use R setup functions

## Checklist

**Before review**

- [x] Update and regenerate man pages
- [ ] Add/update tests
- [x] Add/update examples in vignettes
- [x] Pass CI checks

**Before merge**

- [ ] Update architecture vignette
- [x] Update development vignette
- [ ] Update features in `README`
- [x] Update `CHANGELOG`

### CODE REVIEW

**Review by rcannood — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,21 @@
+# laminr v0.3.0
+
+## NEW FUNCTIONALITY
+
+- Add a `install_lamindb()` function to help with setting up the default Python environment
+- Add `lamin_login()` and `lamin_connect()` functions to allow access to CLI functionality from R
```
Comment (falexwolf) [inline CHANGELOG.md:6]: Cool!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,21 @@
+# laminr v0.3.0
+
+## NEW FUNCTIONALITY
+
+- Add a `install_lamindb()` function to help with setting up the default Python environment
```
Comment (falexwolf) [inline CHANGELOG.md:5]: Cool!

File: DESCRIPTION [hunk 3]
```diff
@@ -23,6 +23,7 @@ Imports:
     purrr,
     R.utils,
     R6,
+    reticulate,
```
Comment (falexwolf) [inline DESCRIPTION:26]: Good!

File: README.md [hunk 4]
```diff
@@ -51,6 +45,36 @@ This will also install these package for the following tasks:
...
+2. Log in
+
+```r
+laminr::lamin_login(api_key = "your_api_key")
+```
+
+OR
+
+```r
+laminr::lamin_login(user = "user_handle")
```
Comment (falexwolf) [inline README.md:67]: This is not going to work, so, I'd remove it. It makes things look complicated.

File: vignettes/setup.Rmd [hunk 5]
```diff
@@ -0,0 +1,133 @@
+---
```
Comment (falexwolf) [inline vignettes/setup.Rmd:1]: Nice! I'll likely amend this slightly at some point but it reads well already! Thank you!

**Review by falexwolf — CHANGES_REQUESTED** [review]
This is great! ❤️ 

Just one mini change request for the docs! :)

File: README.md (hunk 4 above)
Comment (falexwolf) [reply README.md:67]: I meanwhile understand that you mention it for switching users that had already been authenticated before. That works but is a total edge case that shouldn't be on the initial tutorial. So, I'm still voting for removing it here.

Comment (lazappi) [reply README.md:67]: As in, this exact code won't work (and just this needs updating) or switch users with a user name will never work (and I need to update the function)?

Comment (falexwolf) [reply README.md:67]: The code will work and it's well explained in the `setup` guide. But it's an edge case that shouldn't show up in the first guide a user encounters who just wants to get going fast without reading through the entire `setup` instruction.

**Pull request merged**